collections/brandedUserProfile.js

```diff
@@ -314,30 +349,38 @@ BrandedUserSchema = new SimpleSchema({
     improvement: fields.long_string(o), // internal.orientation.improvement
     willYouAttend: fields.string(o), // internal.orientation.willYouAttend
     registered: fields.bool(o) // internal.registeredOrientation
-  }), optional: true},
+  }),
+  optional: true},
   _parking: {type: new SimpleSchema({
     bringCar: fields.bool(o) // _internal.needsParking
-  }), optional: true},
+  }),
+  optional: true},
   _phone: fields.string(o),
   _previousPhone: fields.string(o),
   _profile: {type: new SimpleSchema({
     description: fields.profile_description(o),
     parent: fields.bool(o),
     plannedApplication: fields.planned_application(o),
     setCanTextFalse: fields.bool(o)
-  }), optional: true},
+  }),
+  optional: true},
   _responseBlackBox: {
     type: Object,
     blackbox: true,
     optional: true
   }
-});
+})
 
 BrandedUserProfiles = new Mongo.Collection('brandedUserProfiles')
 BrandedUserProfiles.attachSchema(BrandedUserSchema)
 
+const DATED_FIELDS = {
+  '_contactSettings.canText': '_contactSettings.canTextLastModified',
+  'holoceneContext.currentContext': 'holoceneContext.lastMessageDate'
+}
+
 BrandedUserProfiles.before.update((userId, doc, fieldNames, modifier) => {
```

Comment:
Just out of curiosity, is there a way to look at the schema itself to determine which fields are date fields? If you did it this way you wouldn't have to maintain a separate collection of keys for fields of type `Date`

Reply:
@djrochford  https://github.com/aldeed/simple-schema-js#defaultvalue

Reply:
^You might be able to set a default in the schema and then run schema.clean()

Reply:
Possibly more relevant https://github.com/aldeed/simple-schema-js#autovalue

Reply:
The collection is there to co-ordinate between fields that get changed and fields that record when that change occurs; not to, in general, keep track of date fields. I'm using it in the code below.

Reply:
I've tried to make that clearer in the latest push.